### ISSUE TITLE
feat: Apply rules only to classes that extends Construct

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#54]
### Feature Description

Currently, the rule applies to classes that do not inherit from the Construct class, but we believe this is not appropriate.  
Therefore, we would like the rule to apply only to classes that extend Construct and Stack classes.

### Motivation

To apply the rules more accurately

### Invalid Code Examples

```ts

export class MyConstruct extends Construct {
  public bucket: IBucket
  // ...
}
```

### Valid Code Examples

```typescript

export  class MyConstruct  {
  public bucket: IBucket
  // ...
}
```

### Rule Type

problem (possible errors)

### Implementation Details

- [X] This rule is related to AWS CDK best practices
- [X] I'm willing to submit a PR for this feature
